NFig/SettingsFactory.cs

```diff
@@ -1214,6 +1227,11 @@ PropertyToSettingDelegate GetPropertyToSettingDelegate(Type type)
 
         delegate TValue SettingGetterDelegate<TValue>(TSettings settings);
 
+        static FieldInfo GetGetterOnlyBackingField(PropertyInfo property)
+        {
+            return property.DeclaringType.GetField("<" + property.Name + ">k__BackingField", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
That was my first thought too, but the backing field name is actually generated by the C# compiler, not the CLR, so in theory it would be consistent across the Framework and Core. That is... unless Roslyn decides to change the convention. It'd be great if this were defined in the C# standard, but I'm guessing it's considered an implementation detail that is left up to the implementer. I wonder if Mono follows the same convention... not that I particularly care about supporting Mono.

I'm going to think about this a little more. I like the usability of it though.